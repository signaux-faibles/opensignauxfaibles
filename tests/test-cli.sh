#!/bin/bash

# Test de bout en bout du CLI et de la documentation de ses commandes.
# Ce script doit être exécuté depuis la racine du projet. Ex: par test-all.sh.

# Setup
FLAGS="$*" # the script will update the golden file if "--update" flag was provided as 1st argument
TMP_DIR="tests/tmp-test-execution-files"
OUTPUT_FILE="${TMP_DIR}/test.output.txt"
GOLDEN_FILE="tests/output-snapshots/test-cli.golden.txt"
mkdir -p "${TMP_DIR}"

echo "" > "${OUTPUT_FILE}"

function test {
  CMD="./$1"
  echo "- ${CMD}"
  echo "$ ${CMD}"  >> "${OUTPUT_FILE}"
  echo "$(${CMD})" >> "${OUTPUT_FILE}"
  echo "---"       >> "${OUTPUT_FILE}"
}

# run test cases
test "sfdata"
test "sfdata --help"
test "sfdata purge"
test "sfdata check"
test "sfdata pruneEntities"
test "sfdata import"
test "sfdata validate"
test "sfdata compact"
test "sfdata reduce"
test "sfdata public"

set -e # will stop the script if any command fails with a non-zero exit code
tests/helpers/diff-or-update-golden-master.sh "${FLAGS}" "${GOLDEN_FILE}" "${OUTPUT_FILE}"

rm -rf "${TMP_DIR}"
