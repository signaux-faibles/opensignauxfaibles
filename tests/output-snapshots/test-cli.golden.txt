
$ ./sfdata
usage: sfdata <command> [--boolean-flag] [--parameter=<value1>,<value2>,...]

Supported commands:

   compact          Compacte la base de données
   reduce           Calcule les variables destinées à la prédiction
   etablissements   Exporte la liste des établissements
   entreprises      Exporte la liste des entreprises
   purge            Supprime une partie des données compactées
   check            Vérifie la validité d'un batch avant son importation
   pruneEntities    Compte/supprime les entités hors périmètre
   import           Importe des fichiers
   validate         Liste les entrées de données invalides
   public           Génère les données destinées au site web

Error: You must pass a command
---
